File: output/ChatGPT/codenet/Java/C++/s893190600.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <unordered_map>

using namespace std;

void solve() {
    int H, W;
    cin >> H >> W;
    vector<vector<char>> C(H + 2, vector<char>(W + 2));
    for (int i = 0; i < H; i++) {
        for (int j = 0; j < W; j++) {
            cin >> C[i + 1][j + 1];
        }
    }
    if (H == 1 && W == 1) {
        cout << "No" << endl;
        return;
    }
    for (int i = 1; i <= H; i++) {
        for (int j = 1; j <= W; j++) {
            if (C[i][j] == '#' && C[i - 1][j] != '#' && C[i + 1][j] != '#' && C[i][j - 1] != '#' && C[i][j + 1] != '#') {
                cout << "No" << endl;
                return;
            }
        }
    }
    cout << "Yes" << endl;
}

int main() {
    solve();
    return 0;
}

File: output/ChatGPT/codenet/Java/Python/s893190600.py
class Solver:
	def __init__(self, in, out):
		self.in = in
		self.out = out

	def solve(self):
		H = self.ni()
		W = self.ni()
		C = []
		for i in range(H+2):
			C.append([0]*(W+2))
		for i in range(H):
			c = list(self.ns())
			for j in range(W):
				C[i+1][j+1] = c[j]
		if H == 1 and W == 1:
			self.prn("No")
			return
		for i in range(1, H+1):
			for j in range(1, W+1):
				if C[i][j] == '#' and C[i-1][j] != '#' and C[i+1][j] != '#' and C[i][j-1] != '#' and C[i][j+1] != '#':
					self.prn("No")
					return
		self.prn("Yes")

	def prn(self, s):
		self.out.println(s)

	def ni(self):
		return int(self.in.nextLine())

	def ns(self):
		return self.in.nextLine()

	def nd(self):
		return float(self.in.nextLine())

	def nl(self):
		return long(self.in.nextLine())

	def ndi(self, n):
		return [int(x) for x in self.in.nextLine().split()]

	def nds(self, n):
		return self.in.nextLine().split()

	def ndd(self, n):
		return [float(x) for x in self.in.nextLine().split()]

	def ndl(self, n):
		return [long(x) for x in self.in.nextLine().split()]

	def nddl(self, n